lib/modules/showImages.js

```diff
@@ -638,292 +572,220 @@ function domainToModuleName(hostname) {
 	return domainStack.pop();
 }
 
-async function checkElementForImage(elem, selfTextExpando = false) {
-	if (module.options.hideNSFW.value) {
-		elem.NSFW = new Thing(elem).isNSFW();
-	} else {
-		elem.NSFW = false;
-	}
-
-	elem.selfTextExpando = selfTextExpando;
-	elem.linklistExpando = !!elem.parentNode.classList.contains('title');
-
-	const href = elem.href;
-	if ((!elem.classList.contains('imgScanned') && (typeof imagesRevealed[href] === 'undefined' || !module.options.ignoreDuplicates.value || isCurrentSubreddit('dashboard')) && href !== null)) {
-		elem.classList.add('imgScanned');
-		dupeAnchors++;
-
-		let detectResult;
-
-		const siteModule = [
-			'default',
-			'defaultVideo',
-			'defaultAudio',
-			domainToModuleName(elem.hostname),
-		]
-			.map(modId => domainModuleMap[modId])
-			.find(mod => mod && (detectResult = mod.detect(elem)));
-
-		if (!siteModule) return;
-
-		elem.site = siteModule.moduleID;
+async function convertGifToVideo(options) {
+	try {
+		const info = await ajax({
+			type: 'json',
+			url: '//upload.gfycat.com/transcodeRelease',
+			data: { fetchUrl: options.src },
+			cacheFor: DAY,
+		});
 
-		if (detectResult && !elem.NSFW) {
-			imagesRevealed[href] = dupeAnchors;
+		if (!info.gfyName) throw new Error();
 
-			try {
-				const options = await siteModule.handleLink(href, detectResult);
-				await createImageExpando(elem, options);
-			} catch (e) {
-				console.error(`showImages: error detecting image expando for ${elem.href}`);
-				console.error(e);
-			}
-		}
-	} else if (!elem.classList.contains('imgScanned')) {
-		const textFrag = document.createElement('span');
-		textFrag.setAttribute('class', 'RESdupeimg');
-		removeRedditExpandoButton(elem);
-		$(textFrag).html(`<a class="noKeyNav" href="#img${parseInt(imagesRevealed[href], 10)}" title="click to scroll to original">[RES ignored duplicate link]</a>`);
-		$(elem).after(textFrag);
+		return {
+			options: await gfycat.handleLink('', [], info),
+			siteModule: gfycat,
+		};
+	} catch (e) {
+		throw Error('Could not convert gif to video', options.src);
```

Comment:
I don't think it's standard for the `Error` constructor to accept multiple arguments, so concat `options.src` with the message.

Also minor nit: `s/throw Error/throw new Error/`